charts/k8s-monitoring/docs/ScrapeApplicationMetrics.md

```diff
@@ -0,0 +1,205 @@
+# Scraping Metrics from an Application
+
+If you have an application running on your Kubernetes cluster, that is exporting metrics you can easily extend the
+configuration in this chart to scrape and forward those metrics.
+
+When adding new configuration, it's helpful to think of it in four phases:
+1. Discovery - How should the collector find my service?
+2. Scraping - How should metrics get scraped from my service?
+3. Processing - Is there any work that needs to be done to these metrics?
+4. Delivery - Where should these metrics be sent?
+
+We will go deeper into each phase below.
+
+## Discovery
+
+The discovery phase is about finding the specific pod or service that needs to be scraped for metrics.
+
+This chart automatically creates three components that you can utilize:
+
+* `discovery.kubernetes.nodes` - Discovers all nodes in the cluster
+* `discovery.kubernetes.pods` - Discovers all pods in the cluster
+* `discovery.kubernetes.services` - Discovers all services in the cluster
+
+These are all [`discovery.kubernetes`](https://grafana.com/docs/agent/latest/flow/reference/components/discovery.kubernetes/)
+components, which gather all the specific resources, using the Kubernetes API. From here, we want to refine the search to just the service or the pod that we want.
+
+### Service discovery
+
+Since you don't want to scrape every service in your cluster, you will use rules to select your specific service based
+on its name, namespace, labels, port names or numbers, and many other variables.
+This is done by using a [`discovery.relabel`](https://grafana.com/docs/agent/latest/flow/reference/components/discovery.relabel/)
+component and adding one or more rules, using special meta-labels that are set automatically by the
+`discovery.kubernetes` component.
+
+Here is an example that filters to a service named "database", in the namespace "blue", with the port named "metrics":
```

Comment:
Did I read that the port must be a string and not a port number... could be worth clarifying

Reply:
Yep. In K8s, ports can have names. Furthermore, on multiport services, [names are required](https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services). The discovery.kubernetes automatically sets both `__meta_kubernetes_service_port_name` and `__meta_kubernetes_service_port_number`, and you need to make sure you use the right one.

Reply:
Let me use the other discovery example to call out a port number